src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/LineWrappingHandler.java

```diff
@@ -169,35 +151,135 @@ public void checkIndentation(DetailAST firstNode, DetailAST lastNode, int indent
         else {
             firstNodeIndent = startIndent;
         }
-        final int currentIndent = firstNodeIndent + indentLevel;
-
         for (DetailAST node : firstNodesOnLines.values()) {
-            final int currentType = node.getType();
+            // ArrayInitIndentation is used for array elements
+            // see https://github.com/checkstyle/checkstyle/issues/9086
+            if (isArrayInitElement(node)) {
+                continue;
+            }
+
+            if (!lineStartsWithComment(node)) {
+                checkFirstNodeOnLine(node, indentLevel, previousLine, firstNodeIndent);
+            }
```

Comment:
Why you are handling commments seperately, any reason for  this change.

Reply:
It is for class definitions, method definitions and fields like https://github.com/pmd/pmd/blob/81739da5caff948dbcd2136c17532b65c726c781/pmd-core/src/main/java/net/sourceforge/pmd/properties/AbstractMultiPackagedProperty.java#L23 If line starts with C-style comment, checkstyle should not check the indentation of the node but the indentation of the comment (which is done in CommentsIndentationCheck).